### ISSUE TITLE
Unsearchable ash pile CTD

### ISSUE THREAD

**Issue by BrotherErryn** [SteveTownsend/SmartHarvestSE#217]
Took a bit to figure out what was causing this one (version 3.3). For some reason there was an ash pile from a Vigilant of Stendarr that looked to be bugged. Looking at it displayed the normal Search prompt, but that didn't work. When SmartHarvest tried to loot from it, instant CTD. The .NET crash log seemed to just indicate that the CTD indeed involved an ash pile. I've seen ash piles do that on rare occasion for years...is there a way SHSE can detect that state and ignore those?

[Crash_2020_9_13_10-48-47.txt](https://github.com/SteveTownsend/SmartHarvestSE/files/5214493/Crash_2020_9_13_10-48-47.txt)

**Comment by SteveTownsend** [comment]
Please also attach _SmartHarvestSE.log_ if you still have it

**Comment by SteveTownsend** [comment]
The crash is at the second line here. Probably need to avoid deref of refr if we don't trust it based on lootability value.
```
			lootability = ValidateTarget(refr, possibleDupes, dryRun, glowOnly);
			if (refr->GetFormType() != RE::FormType::ActorCharacter && !refr->GetContainer())
```

**Comment by SteveTownsend** [comment]
We have to check _refr_ is valid before using it. The return code from _ValidateTarget_ per crash dump is 0x35 -> _CannotLootAshPile_, so this should all work if the invalid _refr_ is not used.

**Comment by BrotherErryn** [comment]
Ah, sorry, no longer have the _SmartHarvestSE.log_ for that crash. Looks like you've got a line on it though.

**Comment by SteveTownsend** [comment]
fixed in v3.3.1.0
